core/simulations/steps/createVaultForCollateral.ts

```diff
@@ -0,0 +1,160 @@
+    await contract.approve(addressJoin, MAX.toFixed(0));
+    console.info('Max allowance given out');
+};
+const extractCollateralFromVat = async (
+    collateralType: CollateralType,
+    collateralOwned: BigNumber,
+    tokenContractAddress: string
+) => {
+    console.info('Extracting collateral');
+    const addressJoin = await getContractAddressByName(TEST_NETWORK, getJoinNameByCollateralType(collateralType));
+    const joinContractHasSufficientFunds = await isBalanceGreaterThan(
+        TEST_NETWORK,
+        tokenContractAddress,
+        addressJoin,
+        collateralOwned
+    );
+    if (!joinContractHasSufficientFunds) {
+        throw new Error('Join contract does not have sufficient funds');
+    }
+    await withdrawCollateralFromVat(TEST_NETWORK, HARDHAT_PUBLIC_KEY, collateralType, undefined);
+    return tokenContractAddress;
+};
+const ensureBalance = async (tokenContractAddress: string, decimals: number, collateralOwned: BigNumber) => {
+    const token = await getErc20Contract(TEST_NETWORK, tokenContractAddress);
+    const balanceHex = await token.balanceOf(HARDHAT_PUBLIC_KEY);
+    const balance = new BigNumber(balanceHex._hex).shiftedBy(-decimals);
+    if (!balance.eq(collateralOwned)) {
+        throw new Error(
+            `Unexpected wallet balance. Expected ${collateralOwned.toFixed()}, Actual ${balance.toFixed()}`
+        );
+    }
+    console.info(`Wallet ${HARDHAT_PUBLIC_KEY} has ${collateralOwned.toFixed()} of token ${tokenContractAddress}`);
+};
+const addCollateralToVault = async (vaultId: number, collateralOwned: BigNumber) => {
+    console.info('Adding collateral to Vault');
+    const vault = await fetchVault(TEST_NETWORK, vaultId);
+    const drawnDebtExact = collateralOwned.multipliedBy(vault.minUnitPrice).dividedBy(vault.stabilityFeeRate);
+    const drawnDebt = new BigNumber(drawnDebtExact.toPrecision(drawnDebtExact.e || 0 + 1, BigNumber.ROUND_DOWN));
+    await changeVaultContents(TEST_NETWORK, vaultId, drawnDebt, collateralOwned);
+    const vaultWithContents = await fetchVault(TEST_NETWORK, vaultId);
+    console.info(
+        `Vault's contents: ${vaultWithContents.collateralAmount.toFixed()} of collateral, ${
+            vaultWithContents.initialDebtDai
+        } of debt`
+    );
+};
+export const getCollateralAmountInVat = async (collateralType: CollateralType) => {
+    const contract = await getContract(TEST_NETWORK, 'MCD_VAT');
+    const typeHex = ethers.utils.formatBytes32String(collateralType);
+    const { dust } = await contract.ilks(typeHex);
+    const { minUnitPrice } = await fetchVaultCollateralParameters(TEST_NETWORK, collateralType);
+    const minCollateralInVault = new BigNumber(new BigNumber(dust._hex).toFixed(0))
+        .shiftedBy(-RAD_NUMBER_OF_DIGITS)
+        .div(minUnitPrice);
+
+    await checkAvailableDebtForAmountAndMinUnitPrice(collateralType, minCollateralInVault, minUnitPrice);
+    return new BigNumber(minCollateralInVault.multipliedBy(1.1).toFixed(0));
```

Comment:
Why is it multiplied by `1.1`?

Reply:
to not be the minimal. But it could. removed.